union-ai-admin/aws/union-ai-admin-role.template.yaml

```diff
@@ -374,6 +374,34 @@ Resources:
               - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/containerinsights/opta-*/host:log-stream:*'
               - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/containerinsights/opta-*/application:log-stream:fluentbit-kube.var.log.containers.union-operator-*'
               - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/containerinsights/opta-*/application:log-stream:fluentbit-kube.var.log.containers.flytepropeller-*'
+          - Sid: 'UnionImageBuilderRepoAdmin'
+            Effect: Allow
+            Action:
+              - ecr:CreateRepository
+              - ecr:DeleteRepository
+              - ecr:TagResource
+              - ecr:UntagResource
+              - ecr:PutLifecyclePolicy
+              - ecr:DeleteLifecyclePolicy
+              - ecr:PutImageTagMutability
+              - ecr:PutImageScanningConfiguration
+              - ecr:BatchDeleteImage
+              - ecr:DeleteRepositoryPolicy
```

Comment:
Curious if we actually need some of these permissions or whether or not they could be scoped down.

Reply:
`Tag`, `Policy`, and `Lifecycle` all are currently necessary or will be for feature management.

I think for some others like `DescribeImages` arguable not but I don't necessary trust Terraform + AWS compatibly. There tend to be a lot of touch points and underlying behavior. 

I am inclined to leave them and isolate the `Resource` to `arn:aws:ecr:*:${AWS::AccountId}:repository/union/*` so we only have these permissions to `union` prefixes repositories